tests/chainer_tests/functions_tests/pooling_tests/test_max_pooling_nd.py

```diff
@@ -0,0 +1,199 @@
+import unittest
+
+import functools
+import itertools
+import math
+import mock
+import numpy
+from operator import mul
+import six
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+from chainer.utils import conv
+
+
+def pooling_patches(dims, ksize, stride, pad, cover_all):
+    """Return tuples of slices that indicate pooling patches."""
+    # Left-top indeces of each pooling patch.
+    if cover_all:
+        xss = itertools.product(
+            *[six.moves.range(-p, d + p - k + s, s)
+              for (d, k, s, p) in zip(dims, ksize, stride, pad)])
```

Comment:
Strictly speaking, if it is better to use `six.moves.range`, it is also better to use `six.moves.zip` as well (ref. [here](https://pythonhosted.org/six/#module-six.moves)). But it is not critical because it is not expected to affect the performance so much.